bokehjs/test/unit/models/formatters/datetime_tick_formatter.ts

```diff
@@ -305,12 +329,12 @@ describe("DatetimeTickFormatter", () => {
     it("should handle above", () => {
       const formatter = new dttf.DatetimeTickFormatter({context: "%Y", context_location: "above"})
       const labels = formatter.doFormat([t, t+HOUR, t+HOUR*2], {loc: 0})
-      expect(labels).to.be.equal(["2022\n06/23", "01h", "02h"])
+      expect(labels).to.be.equal(["2022\n06/23", "\n01h", "\n02h"])
     })
     it("should handle below", () => {
       const formatter = new dttf.DatetimeTickFormatter({context: "%Y", context_location: "below"})
       const labels = formatter.doFormat([t, t+HOUR, t+HOUR*2], {loc: 0})
-      expect(labels).to.be.equal(["06/23\n2022", "01h", "02h"])
+      expect(labels).to.be.equal(["06/23\n2022", "01h\n", "02h\n"])
```

Comment:
They ensure that the context & the base ticks stays within the same "row". Using `context_location = "above"` the current behavior causes visual shifts which I think is not desired. Note: This can also happen with nested contexts.
Stripping the newlines at the beginning or end would therefore require additional logic accounting for all corner cases. 
![image](https://github.com/bokeh/bokeh/assets/71090451/a0cd4807-00a9-4360-8fd6-a60ec8adede7)